### PR TITLE
[EX-3846] Add logger

### PR DESCRIPTION
**Pull request by pmareke**
Import the `logger` package inside the new mono-repository

### CODE REVIEW

File: packages/logger/tslint.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (javieri-empathy) [inline packages/logger/tslint.json:1]: Remove this file as it is no longer necessary

File: packages/logger/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+npmPackagePipeline {
```
Comment (javieri-empathy) [inline packages/logger/Jenkinsfile:1]: Remove this file as it is no longer necessary

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/logger/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "@empathyco/x-logger",
+  "version": "1.1.0",
+  "description": "Empathy simple logger",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./types/index.d.ts",
+	"publishConfig": {
```
Comment (javieri-empathy) [inline packages/logger/package.json:8]: Format this new block

**Review by pmareke — COMMENTED** [review]
Fixed all the comments

**Review by javieri-empathy — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -72,6 +72,8 @@ perfect components mixing. A project ready to connect to any Search API with cus
 * [@empathyco/eslint-plugin-x](https://github.com/empathyco/x/tree/main/packages/eslint-plugin-x) - The package name is just to follow the 
 guidelines of ESLint to develop a custom plugin, but it includes also style formatter and good practices outside ESLint, as 
 [prettier](https://prettier.io/) and [standard-version](https://github.com/conventional-changelog/standard-version) 
+* [@empathyco/x-logger](https://github.com/empathyco/x/tree/main/packages/x-logger) - This is package is a custom implementation of the
+default node logger.
```
Comment (LuisMartinez15) [inline .github/CONTRIBUTING.md:76]:
```suggestion
* [@empathyco/x-logger](https://github.com/empathyco/x/tree/main/packages/x-logger) - A custom implementation of default node's logger.
```

File: packages/logger/package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "type": "git",
+    "url": "git+https://cbn-javieri@bitbucket.org/colbenson/logger.git"
+  },
+  "keywords": [
+    "search",
+    "types",
+    "models",
+    "interfaces"
+  ],
+  "author": "Empathy",
```
Comment (LuisMartinez15) [inline packages/logger/package.json:26]: Shouldn't we use the same author as in the `eslint-plugin` package? There we defined it as:
` "author": "Empathy Systems Corporation S.L.",`

File: packages/logger/.gitignore [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+yarn-error.log*
+
+# Editor directories and files
+.idea
+.vscode
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw*
```
Comment (LuisMartinez15) [inline packages/logger/.gitignore:25]: Does it makes sense to have a `.gitignore` file per package? We already have one at `root` level. Just thinking out loud here.

File: packages/logger/package.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  "scripts": {
+    "build": "tsc && tsc --project tsconfig.esm.json",
+    "build:watch": "tsc --watch",
+    "test": "jest"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://cbn-javieri@bitbucket.org/colbenson/logger.git"
+  },
+  "keywords": [
```
Comment (LuisMartinez15) [inline packages/logger/package.json:20]: We might want to update this keywords, this project is not related in any way with these `keywords`.

Suggestions:
- `logger`
- `logging`
- `tools`

File: packages/logger/package.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    "search",
+    "types",
+    "models",
+    "interfaces"
+  ],
+  "author": "Empathy",
+  "license": "ISC",
```
Comment (LuisMartinez15) [inline packages/logger/package.json:27]: Update `license` to `Apache-2.0` as we did for the `eslint-plugin` package.

```suggestion
  "license": "Apache-2.0",
```

File: packages/logger/package.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  "homepage": "https://bitbucket.org/colbenson/logger#readme",
```
Comment (LuisMartinez15) [inline packages/logger/package.json:28]: Update `homepage` url to this repository:

```suggestion
  "homepage": "https://github.com/empathyco/x/tree/main/packages/logger",
```

File: packages/logger/package.json [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  "types": "./types/index.d.ts",
+  "publishConfig": {
+    "access": "public"
+  },
+  "scripts": {
+    "build": "tsc && tsc --project tsconfig.esm.json",
+    "build:watch": "tsc --watch",
+    "test": "jest"
+  },
+  "repository": {
```
Comment (LuisMartinez15) [inline packages/logger/package.json:16]: Is this property still necessary? If so, update the url pointing to this repository.

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

**Review by pmareke — COMMENTED** [review]
Fixed

File: packages/logger/.gitignore (hunk 6 above)
Comment (javieri-empathy) [reply packages/logger/.gitignore:25]: I thought the same, but as each package has its own dependencies and output folder I would leave it there for the moment. In other words, see this `.gitignore` strictly related with the dependencies and the build configuration of each package, which goes in the `package.json`

**Review by javieri-empathy — DISMISSED** [review]

**Review by pmareke — COMMENTED** [review]
Fixed.

**Review by javieri-empathy — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]
